src/sentry/search/utils.py

```diff
@@ -40,6 +40,32 @@ def parse_status_value(value):
     raise ValueError("Invalid status value")
 
 
+def parse_duration(value, interval):
+    try:
+        value = float(value)
+    except ValueError:
+        raise InvalidQuery(u"{} is not a valid duration value".format(value))
+
+    if interval == "ms":
+        delta = timedelta(milliseconds=value)
+    elif interval == "s":
+        delta = timedelta(seconds=value)
+    elif interval == "min":
+        delta = timedelta(minutes=value)
+    elif interval == "hr":
+        delta = timedelta(hours=value)
+    elif interval == "day":
+        delta = timedelta(days=value)
+    elif interval == "wk":
```

Comment:
Based on Adhiraj's feedback at weekly review, it would be good to also support the same duration period names as issue search does. So `w` `h` `d` etc. Having both the issue form and discover share code as well.